packages/module/src/MessageBar/AttachButton.tsx

```diff
@@ -7,14 +7,21 @@ import { forwardRef } from 'react';
 
 // Import PatternFly components
 import { Button, ButtonProps, DropEvent, Icon, Tooltip, TooltipProps } from '@patternfly/react-core';
-import { useDropzone } from 'react-dropzone';
+import { Accept, useDropzone } from 'react-dropzone';
 import { PaperclipIcon } from '@patternfly/react-icons/dist/esm/icons/paperclip-icon';
 
 export interface AttachButtonProps extends ButtonProps {
   /** Callback for when button is clicked */
   onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
   /** Callback function for AttachButton when an attachment is made */
   onAttachAccepted?: (data: File[], event: DropEvent) => void;
+  /** Specifies the file types that the attachment upload component should accept.
+   *  Files that don't match the accepted types will be disabled in the file picker.
+   *  @example
+   *  allowedFileTypes: { 'application/json': ['.json'], 'text/plain': ['.txt'] }
+   *
+   *  **/
+  allowedFileTypes?: Accept;
```

Comment:
I defer to Erin here, but this renders like this:

<img width="409" alt="Screenshot 2025-06-03 at 9 35 21 AM" src="https://github.com/user-attachments/assets/5a204ae9-1deb-4df7-95e6-5b5e5ad4144d" />

We don't usually use `@example`, but I can definitely see these examples being useful here since it's very defined syntax. Do we want to link to the react-dropzone docs at all?